src/components/shareable-url-modal/shareable-url-modal.js

```diff
@@ -125,83 +128,157 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
 
   const handleModalClose = () => {
     onToggleModal(false);
-    setDeploymentState('default');
+    if (deploymentState !== 'incompatible') {
+      setDeploymentState('default');
+    }
     setResponseError(null);
     setIsLoading(false);
     setResponseUrl(null);
     setIsLinkSettingsClick(false);
     setInputValues({});
     setIsFormDirty({
       bucket_name: false,
-      region: false,
+      platform: false,
+      endpoint: false,
     }); /* eslint-disable camelcase */
   };
 
+  /**
+   * Returns the modal heading based on the given type and deployment state.
+   * @param {string} type - The type of the modal heading.
+   * @returns {string|null} The modal heading text or null if deployment state is 'default'.
+   */
+  const getModalHeading = (type) => {
+    if (deploymentState === 'default') {
+      return null;
+    }
+
+    if (type === 'title') {
+      return deploymentState === 'success'
+        ? 'Kedro-Viz Published and Hosted'
+        : 'Publish and Share Kedro-Viz';
+    }
+
+    return modalMessages(deploymentState, compatibilityData.package_version);
+  };
+
+  const handleResponseUrl = () => {
+    // If the URL does not start with http:// or https://, append http:// to avoid relative path issue
+    if (!/^https?:\/\//.test(responseUrl)) {
+      const url = 'http://' + responseUrl;
+      return url;
+    }
+    return responseUrl;
+  };
```

Comment:
I am not sure how to handle this if the user enters a wrong endpoint `url` . There can be many cases of url not entered properly. We should document this in our docs and avoid handling all cases. I think you did this to handle gcp load balancer usecase @jitu5 ?

Reply:
In the case of GCP with a load balancer, the endpoint will most likely be an IP address, such as `32.206.61.13:80`. If a user pastes this IP address into the endpoint input field, the clickable link on the success page becomes relative to the current address, like `https://example.com/32.206.61.13:80`. To address this, I have appended 'http://' to the link.

Should this handling be specific to GCP?